pennylane/ops/qubit/special_unitary.py

```diff
@@ -341,7 +631,76 @@ def compute_decomposition(theta, wires, num_wires):
         [QubitUnitary(array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
             [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]]), wires=[0])]
         """
+        if qml.math.requires_grad(theta):
+            interface = qml.math.get_interface(theta)
+            # Get all Pauli words for the basis of the Lie algebra for this gate
+            words = pauli_words(num_wires)
+
+            # Compute the linear map that transforms between the Pauli basis and effective generators
+            # Consider the mathematical derivation for the prefactor 2j
+            omega = qml.math.real(2j * get_one_parameter_coeffs(theta, num_wires, interface))
+
+            # Create zero parameters for each Pauli rotation gate that take over the trace of theta
+            detached_theta = _detach(theta, interface)
+            zeros = theta - detached_theta
+            # Apply the linear map omega to the zeros to create the correct preprocessing Jacobian
+            zeros = qml.math.tensordot(omega, zeros, axes=[[1], [0]])
+
+            # Apply Pauli rotations that yield the Pauli basis derivatives
+            paulirots = [
+                TmpPauliRot(zero, word, wires=wires, id="SU(N) byproduct")
```

Comment:
Why not use `PauliRot` directly?

Reply:
It makes a lot of sense! It would be nice to have this functionality inside `PauliRot` (such that when the angle is zero, the gate vanishes), but I can't think of an easy way to do it. 

I like this other solution 👍